### DISCUSSION TITLE
.epw files used in UWG

### DISCUSSION THREAD

**Discussion by aleksandraradecka1** [UMEP-dev/UMEP#418] [General]
Hello, 

I am using the UWG module, and in order to make the results more accurate I decided to replace the epw. file downloaded from https://energyplus.net/weather with the file created myself based on ERA5. My colleage prepared the file according to the instructions found in the first tutorial https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/DailyShading.html#dailyshading I am not sure however if the two files - the tutorial file and .epw, containing meteorological data are anyhow comparable. 

I would be very greatful for some advice on this. 
Best regards, 
Aleksandra

**Comment by biglimp** [comment]
I am not sure I follow. There is no meteorological data used in that tutorial...

**Reply by aleksandraradecka1** [reply]
Sorry if I named this wrongly. I meant the file named "GBG_TMY_1977.txt." (inside "Goteborg_SWEREF99_1200.zip")

**Comment by biglimp** [comment]
Ok. Yes, they are comparable in the sense that they boty include meteorological data for one year with hourly resolution. The relevant parameters are also the same (Ta, RH, Kdown, Kdir, Kdiff etc.). What you need to consider also the timing (UTC) the data and the ground temperature in the header. Personally I dont think this is super important but I have not tried out the sensitivity for this parameter.

**Reply by aleksandraradecka1** [reply]
OK, this is a bit relieving for my colleague. The variables' names are different at the first sight, but I understand they mostly describe the same variables. We will work on this. Thank you for your help.

**Comment by aleksandraradecka1** [comment]
Dear Fredrik, 

we are in the process of preparing the new version of .epw file based on ERA5 and one additional question arised: Which of the variables consisted in the file (header info and columns) are anyhow used by UWG algorithm? I of course know that their impact on the final UHI's intensity is still to be assesed. 

I would be greatful for the help.

**Reply by biglimp** [reply]
I think these are the ones (pyid is the python column number, i.e. 6 mean 7th column): 

WEATHER FORCING
var         pyid    num     unit
Temp        6       float   degC
dewpoint    7       float   degC
Relhum      8       float   %
Pressure    9       int     Pa
Ldown       12      float   Wm-2
Kdown       13      float   Wm-2
Kdir        14      float   Wm-2
Kdiff       15      float   Wm-2
Wdir        20      float   deg
Ws          21      float   ms-1

**Reply by aleksandraradecka1** [reply]
Thanks a lot! Does it also mean that the header info may be neglected? I remember you mentioned the ground temperatures a few posts ago and therefore I was prepared for working on the header too. Of course if you do not have the answer at hand please do not bother yourself working on this - I will make the required experiments and find out.

**Reply by biglimp** [reply]
@j3r3m1 , didn't you retreive ground temperature data from ERA5 to be included in an epw-file?

**Reply by j3r3m1** [reply]
Right ! There is indeed 4 levels of soil temperature available in the [ERA5 data](https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcds.climate.copernicus.eu%2Fcdsapp%23!%2Fdataset%2Freanalysis-era5-land%3Ftab%3Doverview&data=04%7C01%7C%7Cfd9c84ad388541a1285b08d9fc4976ec%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637818214725943753%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=ew4y3Od6Rsd7iy0yni4br14yZ53mxjRF9rinJV8%2FMEQ%3D&reserved=0)) :
- level 1 : 0.035 m
- level 2 : 0.175 m
- level 3 : 0.64 m
- level 4 : 1.94 m

From what I have found about EPW, the number after GROUND_TEMPERATURE are well defined in [this page](https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fbigladdersoftware.com%2Fepx%2Fdocs%2F8-3%2Fauxiliary-programs%2Fenergyplus-weather-file-epw-data-dictionary.html&data=04%7C01%7C%7Cfd9c84ad388541a1285b08d9fc4976ec%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637818214725943753%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=5ke1JHQwG%2Btsof19wlXvmZKaNZe3U2%2BwGlQpctFBgAg%3D&reserved=0) (I have put below the needed informations):

GROUND TEMPERATURES,
       N1, Number of Ground Temperature Depths
       N2, \field Ground Temperature Depth 1
         \units m
       N3, \field Depth 1 Soil Conductivity
         \units W/m-K,
       N4, \field Depth 1 Soil Density
         \units kg/m3
       N5, \field Depth 1 Soil Specific Heat
         \units J/kg-K,
       N6, \field Depth 1 January Average Ground Temperature
        \units C
       N7, \field Depth 1 February Average Ground Temperature
        \units C
       N8, \field Depth 1 March Average Ground Temperature
        \units C
       N9, \field Depth 1 April Average Ground Temperature
        \units C
       N10, \field Depth 1 May Average Ground Temperature
        \units C
       N11, \field Depth 1 June Average Ground Temperature
        \units C
       N12, \field Depth 1 July Average Ground Temperature
        \units C
       N13, \field Depth 1 August Average Ground Temperature
        \units C
       N14, \field Depth 1 September Average Ground Temperature
        \units C
       N15, \field Depth 1 October Average Ground Temperature
        \units C
       N16, \field Depth 1 November Average Ground Temperature
        \units C
       N17, \field Depth 1 December Average Ground Temperature
        \units C
      \note repeat above (N2-N17) to number of ground temp depths indicated
-- etc to # of depths entered

I think with this information you should be able to derive the monthly average ground temperature for each level from the ERA5 data and fill the needed "columns in the header". We did not test the influence of this parameter but I am not sure it would affect the results. Let us know if so !

**Reply by aleksandraradecka1** [reply]
Sorry for the late reply. Thank you very much for your help 👍 Of course, after finishinig with the experiments I will let you know about the details (by mail probably).
